lib/csv_export.rb

```diff
@@ -0,0 +1,70 @@
+module CsvExport
+  require 'csv'
+  require 'zip'
+
+  # generates a csv from the given model data
+  def generate_csv(data)
+    objects = data.first
+    columns = data.size == 1 ? objects.first.attributes.keys : data[1]
```

Comment:
I'm not sure I understand why you did it this way. If I'm reading this correctly, `data` is meant to be an array with two elements - an array of database records in the first and then an array of column headers in the second. What case is the ternary operator meant to handle?

Separately, for readability reasons, I'm not sure that I like the use of `data` to contain *both* the actual data and the column headings. It's just a bit confusing; how about having the data array (defaulting to empty) being for the data itself and a second `cols` parameter (also defaulting to empty) for the column headings? Then we could always try to automatically extract the column headings (and remove the `id` column) if it isn't passed explicitly. What do you think?